tensorflow-ops/tests/GradientTest.hs

```diff
@@ -716,10 +716,9 @@ testConv2DBackpropInputGrad = testCase "testConv2DBackpropInputGrad" $ do
         filter' <- TF.render $ TF.fill filterShape (TF.scalar (1::Float))
         let y = TF.conv2DBackpropInput'
                 ( (TF.opAttr "strides" .~ [1::Int64, 1, 1, 1])
-                . (TF.opAttr "padding" .~ (BS.pack "VALID"))
                 . (TF.opAttr "data_format" .~ (BS.pack "NHWC"))
                 )
-                conv_input_shape filter' x
+                "VALID" conv_input_shape filter' x
```

Comment:
Can this "VALID" plumbing be hidden a bit better? Or at least annotated with a comment?
At least link to https://www.tensorflow.org/api_docs/python/tf/nn/convolution.

Reply:
Probably the best solution would be to create a module introduce new types for where "VALID" appears and then have people import said module instead of from `GenOps.Core`.

An example:
```
data Padding = PaddingValid -- ^ output_spatial_shape[i] = ceil(input_spatial_shape[i] / strides[i])
             | PaddingSame  -- ^ output_spatial_shape[i] = ceil((input_spatial_shape[i] - (spatial_filter_shape[i]-1) * dilation_rate[i]) / strides[i])
```
 
Do you agree with this path?

Reply:
I've gone ahead and created a Convolution module that wraps these functions (as well as other convolution functions) to give a better interface.